AirlineBookingApp/FlightManagement/FlightManager.cs

```diff
@@ -0,0 +1,149 @@
+﻿using AirlineBookingApp.DataManagement;
+using AirlineBookingApp.UserManagement;
+using System;
+using System.Collections.Generic;
+using System.Data;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace AirlineBookingApp.FlightManagement
+{
+    internal class FlightManager : UserManagement.Manager
+    {
+        private List<Flight> flights;
+
+        public FlightManager(string username, string password, string email, Role role, string deparment, List<Flight> flights) : base(username, password, email, role, deparment)
+        {
+            this.flights = flights;
+        }
+        public void AddFlight(Flight flight)
+        {
+            flights.Add(flight);
+            Console.WriteLine("The flight has been added successfully!");
+        }
+        public void RemoveFlight(Flight flight)
+        {
+            var flightToRemove = flights.Find(targetFlight => targetFlight.FlightId == flight.FlightId);
+            if (flightToRemove != null)
+            {
+                flights.Remove(flightToRemove);
+                Console.WriteLine("The flight has been deleted successfully!");
+            }
+            else
+            {
+                Console.WriteLine("Flight not Found");
+            }
+        }
+
+        public bool UpdateSeatAvailability(int flightId, string seatClass, int numSeatsToBook)
+        {
```

Comment:
this method is not well implemented, first the name suggests that you update something, but the behavior actually does 2 things, checking and updating, I suggest having a separate function that checks if there is available seat, and in the caller side if that function returns true, you can call the update function that decrements the number of available seats, your code paths should be explicit, and hide only implementation details , not behavior